xonsh/parsers/base.py

```diff
@@ -1894,15 +1895,16 @@ def p_testlist_test(self, p):
         """testlist : test"""
         p1 = p[1]
         if isinstance(p1, ast.Tuple) and (hasattr(p1, '_real_tuple') and
-                                          p1._real_tuple):
+                                          p1._real_tuple and p1.elts):
             p1.lineno, p1.col_offset = lopen_loc(p1.elts[0])
         p[0] = p1
 
     def p_testlist_single(self, p):
         """testlist : test COMMA"""
         p1 = p[1]
-        if isinstance(p1, ast.Tuple) and (hasattr(p1, '_real_tuple') and
-                                          p1._real_tuple):
+        if isinstance(p1, ast.List) or (isinstance(p1, ast.Tuple) and
+                                        hasattr(p1, '_real_tuple') and
+                                        p1._real_tuple):
```

Comment:
`(hasattr(p1, '_real_tuple') and p1._real_tuple)` can be simplified to `getattr(p1, '_real_typle', None)` indeed, but there's same code elsewhere in this file, so i just keep it this way...